src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -844,8 +843,7 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::init_tm_tx(
     auto delay_ms = _metadata_dissemination_retry_delay_ms;
     auto aborted = false;
 
-    auto tx_ntp_opt = _tx_coordinator_ntp_mapper.local().ntp_for(
-      kafka::group_id(tx_id));
+    auto tx_ntp_opt = _tx_coordinator_ntp_mapper.local().ntp_for(tx_id);
```

Comment:
looks dangerous we may forget to change it; it's better to assume that hash ranges are already replicated and try to find a partition based on hosted ranges and fall back to static mapping only if we found nothing